usrp_drivers/n200/n200_driver.cpp

```diff
@@ -24,271 +24,339 @@
+                const DriverOptions &driver_options)
+{
+  DEBUG_MSG("Enter transmit thread");
 
 
-  zmq::socket_t packet_socket(*driver_c, ZMQ_SUB);
-  packet_socket.connect("inproc://threads");
-  packet_socket.setsockopt(ZMQ_SUBSCRIBE, "", 0);
-  std::cout << "TRANSMIT: Creating and connected to thread socket in control\n";
-  zmq::message_t request;
+  zmq::socket_t driver_packet_pub_socket(driver_c, ZMQ_SUB);
+  ERR_CHK_ZMQ(driver_packet_pub_socket.connect("inproc://threads"))
+  ERR_CHK_ZMQ(driver_packet_pub_socket.setsockopt(ZMQ_SUBSCRIBE, "", 0))
+  DEBUG_MSG("TRANSMIT: Creating and connected to thread socket in control");
 
-  zmq::socket_t timing_socket(*driver_c, ZMQ_PAIR);
-  timing_socket.connect("inproc://timing");
+  zmq::socket_t receive_side_timing_socket(driver_c, ZMQ_PAIR);
+  receive_side_timing_socket.connect("inproc://timing");
 
-  zmq::socket_t ack_socket(*driver_c, ZMQ_PAIR);
-  ack_socket.connect("inproc://ack");
+  zmq::socket_t ack_socket(driver_c, ZMQ_PAIR);
+  ERR_CHK_ZMQ(ack_socket.connect("inproc://ack"))
 
 
-  auto channels_set = false;
+  auto usrp_channels_set = false;
   auto center_freq_set = false;
   auto samples_set = false;
 
   std::vector<size_t> channels;
 
-  uhd::time_spec_t time_zero;
-  zmq::message_t timing (sizeof(time_zero));
+  uhd::time_spec_t time_zero; //s
+  zmq::message_t start_receive_timing (sizeof(time_zero));
 
   uhd::tx_streamer::sptr tx_stream;
   uhd::stream_args_t stream_args("fc32", "sc16");
 
   size_t samples_per_buff;
   std::vector<std::vector<std::complex<float>>> samples;
 
-  auto atten_window_time_start_us = driver_options->get_atten_window_time_start();
-  auto atten_window_time_end_us = driver_options->get_atten_window_time_end();
-  auto tr_window_time_us = driver_options->get_tr_window_time();
+  auto atten_window_time_start_s = driver_options.get_atten_window_time_start(); //seconds
+  auto atten_window_time_end_s = driver_options.get_atten_window_time_end(); //seconds
+  auto tr_window_time_s = driver_options.get_tr_window_time(); //seconds
 
   //default initialize SS. Needs to be outside of loop
   auto scope_sync_low = uhd::time_spec_t(0.0);
 
-  auto rx_rate = driver_options->get_rx_rate();
-
-  uint32_t sqn_num = -1;
+  uint32_t sqn_num = 0;
   uint32_t expected_sqn_num = 0;
+// REVIEW #0 If there are large periods of time between pulses, this while loop might be too speedy, resulting in overflows to the usrp - may have seen this in testing? - can we calculate an amount of time to sleep if that's the case?
+// REPLY need to discuss this.
+
+  /*This loop accepts pulse by pulse from the radar_control. It parses the samples, configures the
+   *USRP, sets up the timing, and then sends samples/timing to the USRPs.
+   */
+  while (1) // REVIEW #35 Over 220 lines. are there any candidates for functions? (timing and cout code?)
+  {         // REPLY can discuss
+    zmq::message_t request;
+    driver_packet_pub_socket.recv(&request);
```

Comment:
Maybe we want to poll here instead? If this is sitting and waiting for a long time (i.e. minutes) then something is probably wrong and we should know about it.

Reply:
Could be a good idea. I'll add it as a TODO. I'd like to make sure everything is working correctly first as is.